backends/core/workers.py

```diff
@@ -851,14 +851,28 @@ def _send_batch_hits(self, batch_payload, user_agent='CRMint / 0.1'):
     Raises: MeasurementProtocolException if the HTTP request fails.
     """
     headers = {'user-agent': user_agent}
-    req = requests.post('https://www.google-analytics.com/batch',
-                        headers=headers,
-                        data=batch_payload)
+    if self._debug:
+      for payload in batch_payload.split('\n'):
+        response = requests.post('https://www.google-analytics.com/debug/collect',
+                                 headers=headers,
+                                 data=payload)
+        result = json.loads(response.text)
+        if (not result['hitParsingResult'] or
+            not result['hitParsingResult'][0]['valid']):
+          message = ('Invalid payload ("&" characters replaced with new lines):'
+                     '\n\n%s\n\nValidation response:\n\n%s')
+          readable_payload = payload.replace('&', '\n')
+          self.log_warn(message, readable_payload, response.text)
+    else:
+      response = requests.post('https://www.google-analytics.com/batch',
+                               headers=headers,
+                               data=batch_payload)
 
-    if req.status_code != requests.codes.ok:
-      raise MeasurementProtocolException('Failed to send event hit with status '
-                                         'code (%s) and parameters: %s'
-                                         % (req.status_code, batch_payload))
+      if response.status_code != requests.codes.ok:
+        raise MeasurementProtocolException('Failed to send event hit with '
+                                           'status code (%s) and parameters: %s'
+                                           % (response.status_code,
+                                             batch_payload))
```

Comment:
Can belong to the line above? `% (response.status_code, batch_payload))`

Reply:
@dulacp nice catch :o) Now I can't resist to doing a bit of pylint-ing.